src/tls13.c

```diff
@@ -4641,25 +4821,36 @@ int DoTls13ClientHello(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
         ERROR_OUT(INVALID_PARAMETER, exit_dch);
     }
 
-    /* Legacy protocol version cannot negotiate TLS 1.3 or higher. */
-    if (args->pv.major > SSLv3_MAJOR || (args->pv.major == SSLv3_MAJOR &&
-                                            args->pv.minor >= TLSv1_3_MINOR)) {
-        args->pv.major = SSLv3_MAJOR;
-        args->pv.minor = TLSv1_2_MINOR;
-        wantDowngrade = 1;
-        ssl->version.minor = args->pv.minor;
+#ifdef WOLFSSL_DTLS13
+    if (ssl->options.dtls) {
+        dtls13_do_legacy_version(ssl, &args->pv, &wantDowngrade);
+        dtlsLegacyVersionChecked = 1;
     }
-    /* Legacy version must be [ SSLv3_MAJOR, TLSv1_2_MINOR ] for TLS v1.3 */
-    else if (args->pv.major == SSLv3_MAJOR && args->pv.minor < TLSv1_2_MINOR) {
-        wantDowngrade = 1;
-        ssl->version.minor = args->pv.minor;
+#endif /* WOLFSSL_DTLS13 */
+
+    if (!dtlsLegacyVersionChecked) {
+        if (args->pv.major > SSLv3_MAJOR || (args->pv.major == SSLv3_MAJOR &&
+                                             args->pv.minor >= TLSv1_3_MINOR)) {
+            args->pv.major = SSLv3_MAJOR;
+            args->pv.minor = TLSv1_2_MINOR;
+            wantDowngrade = 1;
+            ssl->version.minor = args->pv.minor;
+        }
+        /* Legacy version must be [ SSLv3_MAJOR, TLSv1_2_MINOR ] for TLS v1.3 */
+        else if (args->pv.major == SSLv3_MAJOR &&
+                 args->pv.minor < TLSv1_2_MINOR) {
+            wantDowngrade = 1;
+            ssl->version.minor = args->pv.minor;
+        }
```

Comment:
Any reason for this to be outside the `WOLFSSL_DTLS13` area?

Reply:
The goal was to reduce the amount of `ifdef`

Reply:
Include this in the `WOLFSSL_DTLS13` section just above it.

Reply:
This part was is needed by tls13. It was used the check to avoid #ifdef #else #endif macro

Reply:
Ok now I see. Please re-instate the following comment before the inner `if` `/* Legacy protocol version cannot negotiate TLS 1.3 or higher. */`